Extra column in a row causes a panic

An extra column in a row causes a panic.  An error message should be reported for this.

```
panic: runtime error: index out of range

goroutine 1 [running]:
panic(0x7fe440, 0xc8200100e0)
        /home/murphyke/go/src/runtime/panic.go:464 +0x3e6
github.com/chop-dbhi/data-models-validator.(*CSVReader).ScanLine(0xc820392480, 0xc82039e000, 0x1d, 0x1d, 0x0, 0x0)
        /home/murphyke/gowork/src/github.com/chop-dbhi/data-models-validator/csv.go:145 +0x3f7
github.com/chop-dbhi/data-models-validator.(*TableValidator).Next(0xc820321ce0, 0x0, 0x0)
        /home/murphyke/gowork/src/github.com/chop-dbhi/data-models-validator/validator.go:157 +0x68
github.com/chop-dbhi/data-models-validator.(*TableValidator).Run(0xc820321ce0, 0x0, 0x0)
        /home/murphyke/gowork/src/github.com/chop-dbhi/data-models-validator/validator.go:197 +0x3d
main.main()
        /home/murphyke/gowork/src/github.com/chop-dbhi/data-models-validator/cmd/validator/main.go:197 +0x14bb
```

--- a/csv.go
+++ b/csv.go
@@ -11,6 +11,7 @@ var (
 	csvErrUnquotedField     = errors.New("unquoted field")
 	csvErrUnescapedQuote    = errors.New("bare quote")
 	csvErrUnterminatedField = errors.New("unterminated field")
+	csvErrExtraColumns      = errors.New("extra columns")
 )
 
 func clearRow(row []string) {
@@ -134,9 +135,17 @@ func (s *CSVReader) Read() ([]string, error) {
 // ScanLine scans all fields in one line and puts the values in
 // the passed slice.
 func (s *CSVReader) ScanLine(r []string) error {
-	var err error
+	var (
+		err error
+		max = len(r)
+	)
 
 	for i := 0; s.Scan(); i++ {
+		// Line too long.
+		if i == max {
+			return csvErrExtraColumns
+		}
+
 		if err = s.Err(); err != nil {
 			clearRow(r[i:])
 			return err

--- a/csv_test.go
+++ b/csv_test.go
@@ -285,6 +285,21 @@ func TestCSVReaderBadInput(t *testing.T) {
 	}
 }
 
+func TestCSVExtraColumns(t *testing.T) {
+	buf := bytes.NewBufferString("one,two,three,four")
+	cr := DefaultCSVReader(buf)
+
+	// 3 columns expected.
+	toks := make([]string, 3)
+	err := cr.ScanLine(toks)
+
+	if err == nil {
+		t.Errorf("Expected error")
+	} else if err != csvErrExtraColumns {
+		t.Errorf("Expected extra columns error, got %s instead", err)
+	}
+}
+
 func BenchmarkCSVReaderScan(b *testing.B) {
 	cr := DefaultCSVReader(&bytes.Buffer{})
 

--- a/validator.go
+++ b/validator.go
@@ -164,6 +164,8 @@ func (t *TableValidator) Next() error {
 			err = ErrUnterminatedColumn
 		case csvErrUnescapedQuote:
 			err = ErrBareQuote
+		case csvErrExtraColumns:
+			err = ErrExtraColumns
 		}
 
 		switch x := err.(type) {

